pkgs/servers/nosql/dragonflydb/default.nix

```diff
@@ -0,0 +1,110 @@
+{ fetchFromGitHub
+, fetchurl
+, lib
+, stdenv
+, double-conversion
+, gperftools
+, mimalloc
+, rapidjson
+, liburing
+, xxHash
+, abseil-cpp_202111
+, gbenchmark
+, glog
+, gtest
+, jemalloc
+, gcc-unwrapped
+, autoconf
+, autoconf-archive
+, automake
+, cmake
+, ninja
+, boost
+, libunwind
+, libtool
+, openssl
+}:
+
+let
+  pname = "dragonflydb";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = "dragonfly";
+    rev = "v${version}";
+    hash = "sha256-P6WMW/n+VezWDXGagT4B+ZYyCp8oufDV6MTrpKpLZcs=";
+    fetchSubmodules = true;
+  };
+
+  # Needed exactly 5.4.4 for patch to work
+  lua = fetchurl {
+    url = "https://github.com/lua/lua/archive/refs/tags/v5.4.4.tar.gz";
+    hash = "sha256-L/ibvqIqfIuRDWsAb1ukVZ7c9GiiVTfO35mI7ZD2tFA=";
+  };
+in
+stdenv.mkDerivation {
+
+  inherit pname version src;
+
+  postPatch = ''
+    mkdir -p ./build/{third_party,_deps}
+    cp -R --no-preserve=ownership ${double-conversion.src} ./build/third_party/dconv
+    cp -R --no-preserve=ownership ${gperftools.src} ./build/third_party/gperf
+    cp -R --no-preserve=ownership ${mimalloc.src} ./build/third_party/mimalloc
+    cp -R --no-preserve=ownership ${rapidjson.src} ./build/third_party/rapidjson
+    cp -R --no-preserve=ownership ${liburing.src} ./build/third_party/uring
+    cp -R --no-preserve=ownership ${xxHash.src} ./build/third_party/xxhash
+    cp -R --no-preserve=ownership ${abseil-cpp_202111.src} ./build/_deps/abseil_cpp-src
+    cp -R --no-preserve=ownership ${gbenchmark.src} ./build/_deps/benchmark-src
+    cp -R --no-preserve=ownership ${glog.src} ./build/_deps/glog-src
+    cp -R --no-preserve=ownership ${gtest.src} ./build/_deps/gtest-src
```

Comment:
Does the build system allow to use symlinks here?

Reply:
I guess it can be done for some, will need to test. Some build in source directory itself so can't be applied to all.

Reply:
Done. Replaced with symlinks for the ones that were possible.